### PR TITLE
feat: Goodbye, node.js Buffer

### PR DESCRIPTION
**Pull request by bjornstar**
Pretty straightforward converting the Buffer usage to Uint8Array.

This PR depends on:
- [x] [`uint8array-extras`](https://github.com/sindresorhus/uint8array-extras/pull/12) as a devDependency.

### CODE REVIEW

**Comment by Borewit** [comment]
Please rebase @bjornstar

File: test/test.ts [hunk 1]
```diff
@@ -88,7 +90,7 @@ for (const tokenizerType of tokenizerTests) {
         const rst = await getTokenizerWithData('\x01\x02\x03\x04\x05\x06', tokenizerType);
         const len = await rst.readBuffer(buffer, {position: 1});
         assert.strictEqual(len, 5, 'return value');
-        assert.strictEqual(uint8ArrayToString(buffer, 'latin1'), '\x02\x03\x04\x05\x06');
```
Comment (bjornstar) [inline test/test.ts]: we added support for various encodings including `latin1` in `uint8array-extras` v1.2.0

**Comment by bjornstar** [comment]
Sorry for force pushing, it's a bad habit ;)

File: test/test.ts (hunk 1 above)
Comment (Borewit) [reply test/test.ts]: Nice, okay, my IDE was telling the second attribute was invalid, maybe old typings somewhere.

Can you you please strip-of my commit and rebase,?
 hope the project is now building on in the PR, after c39057d08bcf179b1b55cdbfd3001def1416628a

**Comment by Borewit** [comment]
> Sorry for force pushing, it's a bad habit ;)

I have no problem with that

File: test/test.ts (hunk 1 above)
Comment (bjornstar) [reply test/test.ts]: Done, just need your approval to run the workflow.

**Comment by bjornstar** [comment]
Everything should be good to go at this point, I'm not sure why Codacy is still showing an error.

**Comment by Borewit** [comment]
Part of [v7.1.0](https://github.com/Borewit/strtok3/releases/tag/v7.1.0)

**Pull request merged**